Crypto/ECC.hs

```diff
@@ -114,7 +118,7 @@ instance EllipticCurve Curve_P256R1 where
         Nothing -> CryptoFailed $ CryptoError_PointSizeInvalid
         Just (m,xy)
             -- uncompressed
-            | m == 4 -> P256.pointFromBinary xy
+            | m == 4 -> P256.pointFromBinary xy >>= validateP256Point
```

Comment:
Might be better to move to `pointFromBinary` being the same function that does the validation internally and maybe expose `unsafePointFromBinary` which does what `pointFromBinary` is doing now.